src/realm/util/file.cpp

```diff
@@ -1064,64 +1064,47 @@ bool File::is_same_file(const File& f) const
 {
     REALM_ASSERT_RELEASE(is_attached());
     REALM_ASSERT_RELEASE(f.is_attached());
+    return f.get_unique_id() == get_unique_id();
+}
 
+File::UniqueID File::get_unique_id() const
+{
+    REALM_ASSERT_RELEASE(is_attached());
 #if REALM_UWP
-    static_cast<void>(f);
-    throw std::runtime_error("Not yet supported");
+    // UWP does not support GetFileInformationByHandleEx(FileIdInfo) or
```

Comment:
GetFileInformationByHandleEx(FileIdInfo) compiles but fails at runtime in the same way as it does on desktop while GetFileInformationByHandle() doesn't compile at all.